### ISSUE TITLE
ZFS send -p does not send properties

### ISSUE THREAD

**Issue by lintonv** [zfsonlinux/zfs#2998]
These are the following issues we are seeing with ZFS send -p

1. For a file system, inital send with a '-p' does not send properties, especially user properties across to the target. BUT with an incremental send with a '-p', the properties seem to go across to the target without errors. Why is there this inconsistency? 

2. For a clone, the same behavior as above, except for the incremental send a bash error of 256 (or 1 on the command line)is returned. This is because zfs_main.c returns (err != 0) which can be true or false. This masks the actual error when using zfs send/recv from the command line. But the properties are send across (again for incremental and not initial)

I wrote a patch that sends only the user properties across to the target and works for all dataset types, whether volume, file system, snapshot or clone. The patch works well for our needs but I need to do more testing on it. If you guys think the patch is necessary or worth looking at, I will post it here,

**Comment by dweeezil** [comment]
For current master code, this should also be fixed by the patch in #2907.  For a simple ```zfs send -p``` of a single snapshot, current master code creates something like this:

```
nvlist version: 0
        tosnap = base
...
                        snapprops = (embedded nvlist)
                        nvlist version: 0
                        (end snapprops)

                (end 0xce16fa8427e05161)

        (end fss)
```
whereas with the patch, we see this:
```
nvlist version: 0
        tosnap = base
...
                        snapprops = (embedded nvlist)
                        nvlist version: 0
                                base = (embedded nvlist)
                                nvlist version: 0
                                (end base)

                        (end snapprops)

                (end 0xce16fa8427e05161)

        (end fss)
```
In this case, I've done a ```zfs send tank/test@base``` but current master code omits the "base=" embedded nvlist within the "snapprops" which causes the properties to not be received properly.

**Comment by lintonv** [comment]
@behlendorf created a pull request - #3001 

@dweeezil Thank you very much for the fix. Just to clarify, the fix is in master but not the 0.6.3-1.2 tag?

**Comment by behlendorf** [comment]
@lintonv Yes, the fix is in master but NOT in 0.6.3-1.2.

**Comment by dweeezil** [comment]
I'm not sure anything has been committed to master yet which would fix this problem.  At least dweeezil/zfs@9533354 from #2907, which would also fix this issue's problem, hasn't been committed yet.

**Comment by behlendorf** [comment]
Whoops, your right.  I thought this has been committed.

**Comment by behlendorf** [comment]
Closing, this should have been addressed by e890dd85a7522730ad46daf68150aafd3952d0c1.
